tests/test_compress_assets_plugin.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding:utf-8 -*-
+
+import os
+
+import blinker
+import pytest
+
+from sigal import init_plugins, signals
+from sigal.gallery import Gallery
+from sigal.plugins import compress_assets
+
+CURRENT_DIR = os.path.dirname(__file__)
+
+
+@pytest.fixture(autouse=True)
+def disconnect_signals():
+    yield None
+    for name in dir(signals):
+        if not name.startswith('_'):
+            try:
+                sig = getattr(signals, name)
+                if isinstance(sig, blinker.Signal):
+                    sig.receivers.clear()
+            except Exception:
+                pass
+
+
+def make_gallery(settings, tmpdir, method):
+    settings['destination'] = str(tmpdir)
+    if "sigal.plugins.compress_assets" not in settings["plugins"]:
+        settings['plugins'] += ["sigal.plugins.compress_assets"]
+
+    # Set method
+    settings.setdefault('compress_assets_options', {})['method'] = method
+
+    compress_options = compress_assets.DEFAULT_SETTINGS.copy()
+    # The key was created by the previous setdefault if needed
+    compress_options.update(settings['compress_assets_options'])
+
+    init_plugins(settings)
+    gal = Gallery(settings)
+    gal.build()
+
+    return compress_options
+
+
+@pytest.mark.parametrize("method,compress_suffix,test_import",
+                         [('gzip', 'gz', None),
+                          ('zopfli', 'gz', 'zopfli.gzip'),
+                          ('brotli', 'br', 'brotli')])
+def test_compress(settings, tmpdir, method, compress_suffix, test_import):
+    if test_import:
+        pytest.importorskip(test_import)
+    compress_options = make_gallery(settings, tmpdir, 'gzip')
```

Comment:
So other methods are not tested because 'gzip' is always used here instead of `method` :sweat_smile:

Reply:
Ahah. You're right. This happens when it's too late to code.